### PR TITLE
GAPI: Move GKernelPackage to cv namespace

### PR DESCRIPTION
**Pull request by alexgiving**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2021.4.1:20.04
build_image:Custom Win=openvino-2021.4.1
build_image:Custom Mac=openvino-2021.4.1

test_modules:Custom=gapi,python2,python3,java
test_modules:Custom Win=gapi,python2,python3,java
test_modules:Custom Mac=gapi,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 1]
```diff
@@ -410,9 +410,13 @@ namespace std
     };
 } // namespace std
 
-
 namespace cv {
+    class GAPI_EXPORTS_W_SIMPLE GKernelPackage;
+
 namespace gapi {
+    GAPI_EXPORTS GKernelPackage combine(const GKernelPackage  &lhs,
+                                        const GKernelPackage  &rhs);
```
Comment (OrestChura) [inline modules/gapi/include/opencv2/gapi/gkernel.hpp:418]: I think it is better to underline they are `cv::GKernelPackage` in this declaration (which is situated in `gapi` namespace, that's why it needs clarification)

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 2]
```diff
@@ -643,9 +649,18 @@ namespace gapi {
          * @param rhs "Right-hand-side" package in the process
          * @return a new kernel package.
          */
-        friend GAPI_EXPORTS GKernelPackage combine(const GKernelPackage  &lhs,
+        friend GAPI_EXPORTS cv::GKernelPackage cv::gapi::combine(const GKernelPackage  &lhs,
```
Comment (OrestChura) [inline modules/gapi/include/opencv2/gapi/gkernel.hpp:652]: and here, as this friend declaration is **actually inside** `GKernelPackage` class, I don't think `cv::` is needed for `cv::GKernelPackage`

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 3]
```diff
@@ -427,7 +431,9 @@ namespace gapi {
     private:
         const char* m_id;
     };
+} // namespace gapi
 
+// namespace cv
```
[the remark this answers is not in the document]
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:436]: I agree that there is no necessity. Removed

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 4]
```diff
@@ -555,7 +561,7 @@ namespace gapi {
          *
          * @param backend backend which kernels to remove
          */
-        void remove(const GBackend& backend);
+        void remove(const gapi::GBackend& backend);
```
[the remark this answers is not in the document]
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:564]: Applied thank you

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 5]
```diff
@@ -643,9 +649,18 @@ namespace gapi {
          * @param rhs "Right-hand-side" package in the process
          * @return a new kernel package.
          */
-        friend GAPI_EXPORTS GKernelPackage combine(const GKernelPackage  &lhs,
+        friend GAPI_EXPORTS cv::GKernelPackage cv::gapi::combine(const GKernelPackage  &lhs,
                                                    const GKernelPackage  &rhs);
```
[the remark this answers is not in the document]
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:653]: Fixed

File: modules/gapi/include/opencv2/gapi/gkernel.hpp (hunk 1 above)
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:418]: Applied

File: modules/gapi/include/opencv2/gapi/gkernel.hpp (hunk 2 above)
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:652]: Applied

File: modules/gapi/src/compiler/passes/passes.hpp [hunk 6]
```diff
@@ -24,10 +24,10 @@ namespace ade {
 }
 
 namespace cv {
+    class GKernelPackage;
 
 // Forward declarations - internal
```
[the remark this answers is not in the document]
Comment (alexgiving) [reply modules/gapi/src/compiler/passes/passes.hpp:29]: Moved, thanks

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 7]
```diff
@@ -500,7 +506,7 @@ namespace gapi {
         }
 
     public:
-        void include(const GFunctor& functor)
+        void include(const gapi::GFunctor& functor)
         {
             m_id_kernels[functor.id()] = std::make_pair(functor.backend(), functor.impl());
         }
```
[the remark this answers is not in the document]
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:512]: Moved, thank you

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 8]
```diff
@@ -632,7 +638,7 @@ namespace gapi {
          *
```
[the remark this answers is not in the document]
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:638]: Moved

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 9]
```diff
@@ -643,9 +642,18 @@ namespace gapi {
          * @param rhs "Right-hand-side" package in the process
          * @return a new kernel package.
          */
-        friend GAPI_EXPORTS GKernelPackage combine(const GKernelPackage  &lhs,
-                                                   const GKernelPackage  &rhs);
+        friend GAPI_EXPORTS GKernelPackage cv::gapi::combine(const GKernelPackage  &lhs,
+                                                             const GKernelPackage  &rhs);
     };
+    /** @} */
+// namespace cv
```
Comment (OrestChura) [inline modules/gapi/include/opencv2/gapi/gkernel.hpp:649]: And here I think it should be removed, too

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 10]
```diff
@@ -595,7 +598,7 @@ namespace gapi {
          *
          */
         template<typename KAPI>
-        GBackend lookup() const
+        gapi::GBackend lookup() const
```
Comment (OrestChura) [inline modules/gapi/include/opencv2/gapi/gkernel.hpp:601]: Here is gapi:: only

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 11]
```diff
@@ -621,18 +624,14 @@ namespace gapi {
...
-        }
+        void include(const cv::gapi::GBackend& backend, const std::string& kernel_id);
 
         /**
          * @brief Lists all backends which are included into package
          *
          * @return vector of backends
          */
-        std::vector<GBackend> backends() const;
+        std::vector<gapi::GBackend> backends() const;
```
Comment (OrestChura) [inline modules/gapi/include/opencv2/gapi/gkernel.hpp:634]: And here, gapi::

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 12]
```diff
@@ -410,9 +410,13 @@ namespace std
     };
 } // namespace std
 
-
 namespace cv {
+    class GAPI_EXPORTS_W_SIMPLE GKernelPackage;
+
 namespace gapi {
+    GAPI_EXPORTS GKernelPackage combine(const cv::GKernelPackage  &lhs,
```
Comment (OrestChura) [inline modules/gapi/include/opencv2/gapi/gkernel.hpp:417]: `cv::GKernelPackage` in output type

File: modules/gapi/include/opencv2/gapi/gkernel.hpp (hunk 9 above)
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:649]: Removed

File: modules/gapi/include/opencv2/gapi/gkernel.hpp (hunk 10 above)
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:601]: Add cv::gapi:: namespace

File: modules/gapi/include/opencv2/gapi/gkernel.hpp (hunk 11 above)
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:634]: Applied

File: modules/gapi/include/opencv2/gapi/gkernel.hpp (hunk 12 above)
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:417]: Applied thanks

**Review by OrestChura — APPROVED** [review]
Everything is great!

1 small alignment comment left

Also, please, ask DM directly about deprecations https://github.com/opencv/opencv/pull/21318#discussion_r774068946

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 13]
```diff
@@ -410,9 +410,13 @@ namespace std
     };
 } // namespace std
 
-
 namespace cv {
+    class GAPI_EXPORTS_W_SIMPLE GKernelPackage;
+
 namespace gapi {
+    GAPI_EXPORTS cv::GKernelPackage combine(const cv::GKernelPackage  &lhs,
+                                        const cv::GKernelPackage  &rhs);
```
[the remark this answers is not in the document]
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:418]: +1

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 14]
```diff
@@ -643,9 +649,18 @@ namespace gapi {
...
          */
-        friend GAPI_EXPORTS GKernelPackage combine(const GKernelPackage  &lhs,
+        friend GAPI_EXPORTS cv::GKernelPackage cv::gapi::combine(const GKernelPackage  &lhs,
                                                    const GKernelPackage  &rhs);
     };
+    /** @} */
+// namespace cv
+
+namespace gapi {
+    using GKernelPackage = cv::GKernelPackage; // Keep backward compatibility
```
[the remark this answers is not in the document]
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:651]: Only after we plan removing `cv::gapi::GKernelPackage`.

**Review by dmatveev — APPROVED** [review]
LGTM. Fix alignment & merge, Thanks

File: modules/gapi/include/opencv2/gapi/gkernel.hpp (hunk 13 above)
[the remark this answers is not in the document]
Comment (alexgiving) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:418]: Fixed, thank you!

File: modules/gapi/include/opencv2/gapi/gkernel.hpp (hunk 14 above)
[the remark this answers is not in the document]
Comment (OrestChura) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp:651]: OK

**Pull request merged**